src/instruments/src/DCDU/index.tsx

```diff
@@ -232,140 +237,132 @@ const DCDU: React.FC = () => {
-    });
-    useCoherentEvent('A32NX_DCDU_ATC_LOGON_MSG', (message: string) => {
-        setAtcMessage(message);
-    });
-    useCoherentEvent('A32NX_DCDU_SYSTEM_ATSU_STATUS', (status: DcduStatusMessage) => {
-        setSystemStatusMessage(status);
-        setSystemStatusTimer(5000);
-    });
-    useCoherentEvent('A32NX_DCDU_MSG_ATSU_STATUS', (uid: number, status: DcduStatusMessage) => {
-        if (!messagesRef.current) {
-            return;
-        }
+                setMessages(newMessageMap);
+            }
+        };
+
+        newSubscriber.on('cpdlcMessages').handle((messages: CpdlcMessage[]) => handleIncomingMessages(messages));
+        newSubscriber.on('dclMessages').handle((messages: DclMessage[]) => handleIncomingMessages(messages));
+        newSubscriber.on('oclMessages').handle((messages: OclMessage[]) => handleIncomingMessages(messages));
+        newSubscriber.on('deleteMessage').handle((uid: number) => closeMessage(uid));
+        newSubscriber.on('logonMessage').handle((message: string) => setAtcMessage(message));
+        newSubscriber.on('systemStatus').handle((status: MailboxStatusMessage) => {
+            setSystemStatusMessage(status);
+            setSystemStatusTimer(5000);
+        });
+        newSubscriber.on('messageStatus').handle((data: { uid: number; status: MailboxStatusMessage }) => {
+            if (!messagesRef.current) {
+                return;
+            }
 
-        const dcduBlock = messagesRef.current.get(uid);
-        if (dcduBlock !== undefined) {
-            dcduBlock.statusMessage = status;
-            if (status === DcduStatusMessage.NoMessage) {
-                if (dcduBlock.messages[0].MessageMonitoring === CpdlcMessageMonitoringState.Monitoring) {
-                    dcduBlock.statusMessage = DcduStatusMessage.Monitoring;
-                } else if (dcduBlock.messages[0].MessageMonitoring === CpdlcMessageMonitoringState.Cancelled) {
-                    dcduBlock.statusMessage = DcduStatusMessage.MonitoringCancelled;
+            const dcduBlock = messagesRef.current.get(data.uid);
+            if (dcduBlock !== undefined) {
+                dcduBlock.statusMessage = data.status;
+                if (data.status === MailboxStatusMessage.NoMessage) {
+                    if (dcduBlock.messages[0].MessageMonitoring === CpdlcMessageMonitoringState.Monitoring) {
+                        dcduBlock.statusMessage = MailboxStatusMessage.Monitoring;
+                    } else if (dcduBlock.messages[0].MessageMonitoring === CpdlcMessageMonitoringState.Cancelled) {
+                        dcduBlock.statusMessage = MailboxStatusMessage.MonitoringCancelled;
+                    }
                 }
+                setMessages(new Map<number, DcduMessageBlock>(messagesRef.current));
             }
-            setMessages(new Map<number, DcduMessageBlock>(messagesRef.current));
-        }
-    });
+        });
+
+        setSubscriber(newSubscriber);
+    }, []);
```

Comment:
We need a way here to unsubscribe from all those things in the cleanup callback, but unfortunately our version of `msfssdk` does not have the `Subscription` type....

Need to think of a way to handle this, or add a TODO. This component will never unmount for now, but it is still unoptimal to leak subscriptions

Reply:
handled the leak in the cleanup-call of the effect